napari/_vispy/canvas.py

```diff
@@ -71,26 +143,258 @@ def _set_theme_change(self, theme: str):
         )[0]
         self.bgcolor = self._last_theme_color
 
+    def _disconnect_theme(self):
+        self.viewer.events.theme.disconnect(self._on_theme_change)
+
     @property
     def bgcolor(self):
-        SceneCanvas.bgcolor.fget(self)
+        return self.scene_canvas.bgcolor.rgba
 
     @bgcolor.setter
     def bgcolor(self, value):
         _value = self._background_color_override or value
-        SceneCanvas.bgcolor.fset(self, _value)
+        self.scene_canvas.bgcolor = _value
 
     @property
     def central_widget(self):
         """Overrides SceneCanvas.central_widget to make border_width=0"""
-        if self._central_widget is None:
-            self._central_widget = Widget(
-                size=self.size, parent=self.scene, border_width=0
+        if self.scene_canvas._central_widget is None:
+            self.scene_canvas._central_widget = Widget(
+                size=self.scene_canvas.size,
+                parent=self.scene_canvas.scene,
+                border_width=0,
             )
-        return self._central_widget
+        return self.scene_canvas._central_widget
 
-    def _process_mouse_event(self, event):
-        """Ignore mouse wheel events which have modifiers."""
-        if event.type == 'mouse_wheel' and len(event.modifiers) > 0:
+    @property
+    def cursor(self) -> QCursor:
+        """Cursor associated with native widget"""
+        return self.scene_canvas.native.cursor()
+
+    @cursor.setter
+    def cursor(self, q_cursor):
+        """Setting the cursor of the native widget"""
+        self.scene_canvas.native.setCursor(q_cursor)
```

Comment:
The logic for creating the QCursor is currently still in the QtViewer due to specific bitmaps being created in case of square and circular cursors.

Reply:
I am not entirely certain whether instead of having this as a property a method would be better. Given that the qt_viewer._on_cursor method uses pixmap functions that are in _qt/utils.py